__tests__/endPoints.test.js

```diff
@@ -1,78 +1,103 @@
-        .expect(404)
-        .then((result)=>{
-            expect(result.body.msg).toBe("404 error")
-        })
-    });
-    describe('GET /api', () => {
-        test('Test for status 200', () => {
-            return superTest(app).get('/api')
-            .expect(200)
-            .then((file)=>{
-                const apiDescription = file.body['GET /api']
-                expect(typeof file.body).toBe('object')
-                expect(Object.keys(apiDescription)).toEqual(['description'])
-                expect(Object.values(file.body['GET /api'])).toEqual(['serves up a json representation of all the available endpoints of the api'])
-            })
-        })
-    });
-    // describe('GET /api/articles/:article_id',()=>{
-    //     test('/api/articles/1',()=>{
-
-    //     })
-    // })
+        return items;
+      })
+      .then((result) => {
+        result.body.forEach((obj) => {
+          expect(obj).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
+        });
+      });
+  });
+  test("Should return an array of all slugs & descriptions", () => {
+    expect(Object.keys(data.topicData[0]).length).toBe(2);
+    expect(data.topicData).toEqual([
+      {
+        description: "The man, the Mitch, the legend",
+        slug: "mitch",
+      },
+      {
+        description: "Not dogs",
+        slug: "cats",
+      },
+      {
+        description: "what books are made of",
+        slug: "paper",
+      },
+    ]);
+  });
+  test("Test for 404 error response", () => {
+    return superTest(app)
+      .get("/api/none-existing-page")
+      .expect(404)
+      .then((result) => {
+        expect(result.body.msg).toBe("404 error");
+      });
+  });
 });
-
-
-
-
-
+describe("GET /api", () => {
+  test("Test for status 200", () => {
+    return superTest(app)
+      .get("/api")
+      .expect(200)
+      .then((file) => {
+        const apiDescription = file.body["GET /api"];
+        expect(typeof file.body).toBe("object");
+        expect(Object.keys(apiDescription)).toEqual(["description"]);
+        expect(Object.values(file.body["GET /api"])).toEqual([
+          "serves up a json representation of all the available endpoints of the api",
+        ]);
+      });
+  });
+});
+describe("GET /api/articles/:article_id", () => {
+  test("Should test for 200 status", () => {
+    return superTest(app).get("/api/articles/2")
+    .expect(200)
+  });
+  test('Should respond with object containing author title, article_id,body,topic,created_at,votes,article_img_url', () => {
+    return superTest(app).get("/api/articles/2")
+    .then((result)=>{
+      console.log(result.body)
+      const objectKeys = Object.keys(result.body[0])
+      const expectedValues = ['article_id','title','topic','author','body','created_at','votes','article_img_url']
+      expect(objectKeys).toEqual(expectedValues)
+      expect(objectKeys.length).toBe(8)
```

Comment:
This method is fine for this task.
Going forward I would take a look at the jest matchers toMatchObject or expect.ObjectContaining they can be very powerful for this sort of testing.